frontend/src/pages/projects/notebook/useRouteForNotebook.ts

```diff
@@ -46,7 +57,7 @@ const useRouteForNotebook = (
       cancelled = true;
       clearTimeout(watchHandle);
     };
-  }, [notebookName, projectName, isRunning]);
+  }, [notebookName, projectName, isRunning, dashboardConfig]);
```

Comment:
Dashboard value may churn due to being a K8s object... the flag for enableServiceMesh is a boolean and thus is a primitive and even if the Dashboard changes the value is unlikely to change so easily.

`obj === obj2` -- even if they have the same values, the two are not equal unless they have the same memory reference. 
`boolA === boolB` -- always will compare value because they are primitives.